### PR TITLE
Resist change to ROB contest

### PR DESCRIPTION
**Pull request by vode-code**
## About The Pull Request
This changes Resisting grabs to be a contest between the assailant's ROB and the grabbed's ROB.

## Why It's Good For The Game
stats good?

## Changelog
:cl: Chickenish
balance: resisting grabs has been changed to a ROB contest.
/:cl:

### CODE REVIEW

**Comment by ValoTheValo** [comment]
does this mean you cant resist out of a grab if you have lower ROB?

**Comment by SirRichardFrancis** [comment]
75 commits?
image

**Comment by vode-code** [comment]
> 75 commits? image

https://git-scm.com/book/en/v2

**Comment by SirRichardFrancis** [comment]
> https://git-scm.com/book/en/v2

Yes. Now read it.

**Comment by vode-code** [comment]
> does this mean you cant resist out of a grab if you have lower ROB?

you can, you just have a lowered chance, capped at either 1% min for aggressive or 0.5% min for neck... ah bleep I just realized killgrabs are entirely inescapable.

**Comment by vode-code** [comment]
> > https://git-scm.com/book/en/v2
> 
> Yes. Now read it.

we should both read it considering that's your best answer.

**Comment by SirRichardFrancis** [comment]
> we should both read it considering that's your best answer.

It's not me who makes PRs with 70+ 8 months old completely unrelated commits.

**Comment by ValoTheValo** [comment]
> > does this mean you cant resist out of a grab if you have lower ROB?
> 
> you can, you just have a lowered chance, capped at either 1% min for aggressive or 0.5% min for neck... ah bleep I just realized killgrabs are entirely inescapable.

does it scale based on how much rob or is it just flat 1% like if im just below the amount of rob of the guy attacking me would it be more?

**Comment by vode-code** [comment]
> does it scale based on how much rob or is it just flat 1% like if im just below the amount of rob of the guy attacking me would it be more?

it scales and 1% is minimum.

**Comment by pexel5** [comment]
Now technomancer who has made the necessary oddity and received an invite 2-3 times will be able to kill an operative in full gear with a grab. not really good.

**Comment by ValoTheValo** [comment]
> Now technomancer who has made the necessary oddity and received an invite 2-3 times will be able to kill an operative in full gear with a grab. not really good.

Well it makes sense for it to be scaled on rob. And if you bulk up on violence or oddities you do deserve to be better in cqc

**Comment by assortedbeads** [comment]
> Now technomancer who has made the necessary oddity and received an invite 2-3 times will be able to kill an operative in full gear with a grab. not really good.

He doesnt know about NT getting overwhelming rob oddities

Acolytes gonna be powerhouses now

**Comment by pexel5** [comment]
> > Now technomancer who has made the necessary oddity and received an invite 2-3 times will be able to kill an operative in full gear with a grab. not really good.
> 
> Well it makes sense for it to be scaled on rob. And if you bulk up on violence or oddities you do deserve to be better in cqc

Not really, die becouse the grab instead of a shootout is not very interesting. you try found weapon, make painkillers and die becouse someone technomancer be traitor and want you head. very funny

**Comment by pexel5** [comment]
> > Now technomancer who has made the necessary oddity and received an invite 2-3 times will be able to kill an operative in full gear with a grab. not really good.
> 
> He doesnt know about NT getting overwhelming rob oddities
> 
> Acolytes gonna be powerhouses now

+

File: code/modules/mob/living/carbon/resist.dm [hunk 1]
```diff
@@ -69,12 +69,12 @@
 			if(GRAB_PASSIVE)
 				qdel(G)
 			if(GRAB_AGGRESSIVE)
-				if(prob(60)) //same chance of breaking the grab as disarm
+				if(prob(max(60 + ((stats?.getStat(STAT_ROB)) - G.assailant?.stats.getStat(STAT_ROB) ** 0.8), 1))) // same scaling as cooldown increase and if you manage to be THAT BAD, 1% for luck
```
Comment (TheShown911) [inline code/modules/mob/living/carbon/resist.dm:72]: You would need to have 0 rob while your attacker has 160 to get down to the 1%

**Review by TheShown911 — APPROVED** [review]

**Comment by TheShown911** [comment]
Overall this is very generous for aggressive grabs, you could be fighting someone with 100 ROB while you only have 30 and still have a >50% chance to escape, and having more rob than someone who is holding you in a neck grab gives you much higher chance to escape.

**Comment by TheShown911** [comment]
Also strange integration failure.

**Comment by SirRichardFrancis** [comment]
> Also strange integration failure.

It's damn mouse again.

File: code/modules/mob/living/carbon/resist.dm (hunk 1 above)
Comment (vode-code) [reply code/modules/mob/living/carbon/resist.dm:72]: that can happen with powergaming ROB.

**Comment by vode-code** [comment]
> Overall this is very generous for aggressive grabs, you could be fighting someone with 100 ROB while you only have 30 and still have a >50% chance to escape, and having more rob than someone who is holding you in a neck grab gives you much higher chance to escape.

Aggressive grabs had a chance starting at 60% so I kept it there with the contest still in place.

**Review by hyperioo — APPROVED** [review]
Merging, it might affect how stat-less mobs escape grabs since they don't have stats to check so a bit of tweaking may be needed next week.

**Comment by hyperioo** [comment]
And please reset your master branch to match eris master branch so that your PRs are not a commit hell anymore.

**Pull request merged**